### PR TITLE
scheduler(engine): fix scheduler panic

### LINKED ISSUE #7670: ScheduleTask panic when there is no executor
### What did you do?

Run a single tiflow master, and then create a job.

### What did you expect to see?

tiflow master should be running as well.

### What did you see instead?

```
panic: invalid argument to Intn

goroutine 484 [running]:
math/rand.(*Rand).Intn(0xc001dcf580?, 0x5d574b0?)
	/home/xyj/.local/lib/go1.19.2/src/math/rand/rand.go:168 +0x65
math/rand.Intn(...)
	/home/xyj/.local/lib/go1.19.2/src/math/rand/rand.go:337
github.com/pingcap/tiflow/engine/servermaster/scheduler.(*Scheduler).ScheduleTask(0xc000d523f0?, {0x5d574b0?, 0xc002017740?}, 0x50cf040?)
	/home/xyj/Projects/tiflow/engine/servermaster/scheduler/scheduler.go:56 +0x46
github.com/pingcap/tiflow/engine/servermaster.(*Server).ScheduleTask(0xc0005b3560, {0x5d574b0, 0xc002017740}, 0x37?)
	/home/xyj/Projects/tiflow/engine/servermaster/server.go:318 +0x108
github.com/pingcap/tiflow/engine/enginepb._TaskScheduler_ScheduleTask_Handler.func1({0x5d574b0, 0xc002017740}, {0x50cf040?, 0xc000d523f0})
	/home/xyj/Projects/tiflow/engine/enginepb/master_grpc.pb.go:431 +0x78
github.com/pingcap/tiflow/engine/pkg/rpcutil.UnaryServerInterceptor({0x5d574b0?, 0xc002017740?}, {0x50cf040, 0xc000d523f0}, 0xc0003fb5a0, 0xc000d24f50?)
	/home/xyj/Projects/tiflow/engine/pkg/rpcutil/middleware.go:100 +0x5a
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x5d574b0?, 0xc002017740?}, {0x50cf040?, 0xc000d523f0?})
	/home/xyj/.local/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1129 +0x5b
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1({0x5d574b0, 0xc002017740}, {0x50cf040, 0xc000d523f0}, 0x50e6480?, 0xc00203e300)
	/home/xyj/.local/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x5d574b0?, 0xc002017740?}, {0x50cf040?, 0xc000d523f0?})
	/home/xyj/.local/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1132 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1({0x5d574b0, 0xc002017740}, {0x50cf040, 0xc000d523f0}, 0xc0003fb5a0, 0xc000d239c8)
	/home/xyj/.local/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1134 +0x12b
github.com/pingcap/tiflow/engine/enginepb._TaskScheduler_ScheduleTask_Handler({0x52c93a0?, 0xc0005b3560}, {0x5d574b0, 0xc002017740}, 0xc000d52380, 0xc001906620)
	/home/xyj/Projects/tiflow/engine/enginepb/master_grpc.pb.go:433 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001b50000, {0x5d7bfe0, 0xc001286ea0}, 0xc002066000, 0xc001b2edb0, 0x7f1c5a0, 0x0)
	/home/xyj/.local/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1295 +0xb2b
google.golang.org/grpc.(*Server).handleStream(0xc001b50000, {0x5d7bfe0, 0xc001286ea0}, 0xc002066000, 0x0)
	/home/xyj/.local/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1636 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/home/xyj/.local/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/xyj/.local/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:930 +0x28a

```

### Versions of the cluster

master

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #7670 

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: engine/servermaster/scheduler/scheduler.go [hunk 1]
```diff
@@ -53,6 +52,9 @@ func (s *Scheduler) ScheduleTask(
 	if err != nil {
 		return nil, err
 	}
+	if len(candidates) == 0 {
+		return nil, errors.ErrNoQualifiedExecutor.FastGenByArgs()
```
Comment (gozssky) [inline engine/servermaster/scheduler/scheduler.go:56]:
```suggestion
		return nil, errors.ErrNoQualifiedExecutor.GenWithStackByArgs()
```

**Review by gozssky — APPROVED** [review]

**Comment by gozssky** [comment]
Please run `make errdoc` to update errors.toml.

**Comment by maxshuang** [comment]
Why we don't trigger this panic before? We don't meet any of this special case?

**Comment by CharlesCheung96** [comment]
> Why we don't trigger this panic before? We don't meet any of this special case?

This problem is caused by newly merged code, ref [pr 7663](https://github.com/pingcap/tiflow/pull/7663/files#diff-5a81fb0246de00b94f2c0e7fd2b11b10a57a2bcbfa33387a06ec75bb38a58082R56)

**Review by amyangfei — APPROVED** [review]

**Pull request merged**